### ISSUE TITLE
No objects detected when using libdarknet-cpp-shared.so and GPU

### ISSUE THREAD

**Issue by Qazi-Farjad** [prabindh/darknet#30]
ubuntu 14.04, cudnn 5.1.10, cuda 8.0, opencv 3.2
GPU = NVIDIA GTX 960
GPU=1
CUDNN=1
When I run ./darknet-cpp detector yolo.cfg yolo.weights.cfg dog.jpg, it runs fine, but when I use the same code (copied test_detector() from detector.c) in a cpp file and use the shared lib, the detector doesnt detect any objects when using GPU. Have tested it with and without cudnn, same results.

Using the shared lib works fine (same code as above) in another laptop where I compiled without GPU.
![scr2](https://cloud.githubusercontent.com/assets/4918481/26391465/58d1fad0-407c-11e7-94a7-2cafe56b0e7f.png)

**Comment by prabindh** [comment]
could you share the gcc command used for building the standalone executable ?

**Comment by Qazi-Farjad** [comment]
g++  -DOPENCV -I/usr/local/include/opencv -I/usr/local/include    -DGPU -I/usr/local/cuda/include/ -DCUDNN  -Wall -Wfatal-errors  -Ofast -DOPENCV -DGPU -DCUDNN ./obj-cpp/gemm.o ./obj-cpp/utils.o ./obj-cpp/cuda.o ./obj-cpp/deconvolutional_layer.o ./obj-cpp/convolutional_layer.o ./obj-cpp/list.o ./obj-cpp/image.o ./obj-cpp/activations.o ./obj-cpp/im2col.o ./obj-cpp/col2im.o ./obj-cpp/blas.o ./obj-cpp/crop_layer.o ./obj-cpp/dropout_layer.o ./obj-cpp/maxpool_layer.o ./obj-cpp/softmax_layer.o ./obj-cpp/data.o ./obj-cpp/matrix.o ./obj-cpp/network.o ./obj-cpp/connected_layer.o ./obj-cpp/cost_layer.o ./obj-cpp/parser.o ./obj-cpp/option_list.o ./obj-cpp/detection_layer.o ./obj-cpp/captcha.o ./obj-cpp/route_layer.o ./obj-cpp/writing.o ./obj-cpp/box.o ./obj-cpp/nightmare.o ./obj-cpp/normalization_layer.o ./obj-cpp/avgpool_layer.o ./obj-cpp/coco.o ./obj-cpp/dice.o ./obj-cpp/yolo.o ./obj-cpp/detector.o ./obj-cpp/layer.o ./obj-cpp/compare.o ./obj-cpp/regressor.o ./obj-cpp/classifier.o ./obj-cpp/local_layer.o ./obj-cpp/swag.o ./obj-cpp/shortcut_layer.o ./obj-cpp/activation_layer.o ./obj-cpp/rnn_layer.o ./obj-cpp/gru_layer.o ./obj-cpp/rnn.o ./obj-cpp/rnn_vid.o ./obj-cpp/crnn_layer.o ./obj-cpp/demo.o ./obj-cpp/tag.o ./obj-cpp/cifar.o ./obj-cpp/go.o ./obj-cpp/batchnorm_layer.o ./obj-cpp/art.o ./obj-cpp/region_layer.o ./obj-cpp/reorg_layer.o ./obj-cpp/lsd.o ./obj-cpp/super.o ./obj-cpp/voxel.o ./obj-cpp/tree.o ./obj-cpp/convolutional_kernels.o ./obj-cpp/deconvolutional_kernels.o ./obj-cpp/activation_kernels.o ./obj-cpp/im2col_kernels.o ./obj-cpp/col2im_kernels.o ./obj-cpp/blas_kernels.o ./obj-cpp/crop_layer_kernels.o ./obj-cpp/dropout_layer_kernels.o ./obj-cpp/maxpool_layer_kernels.o ./obj-cpp/network_kernels.o ./obj-cpp/avgpool_layer_kernels.o ./obj-cpp/darknet.o -o darknet-cpp -lm -pthread  -L/usr/local/lib -lopencv_cudabgsegm -lopencv_cudaobjdetect -lopencv_cudastereo -lopencv_ml -lopencv_shape -lopencv_stitching -lopencv_cudafeatures2d -lopencv_superres -lopencv_cudacodec -lopencv_videostab -lopencv_cudaoptflow -lopencv_cudalegacy -lopencv_calib3d -lopencv_features2d -lopencv_highgui -lopencv_videoio -lopencv_photo -lopencv_imgcodecs -lopencv_cudawarping -lopencv_cudaimgproc -lopencv_cudafilters -lopencv_video -lopencv_objdetect -lopencv_imgproc -lopencv_flann -lopencv_cudaarithm -lopencv_core -lopencv_cudev    -L./3rdparty -L/usr/local/cuda/lib64 -lcuda -lcudart -lcublas -lcurand -lcudnn -lstdc++

**Comment by prabindh** [comment]
No, I was looking for the build command for the standalone application that used darknet-shared lib.

**Comment by salmanmaq** [comment]
g++ -L/usr/local/lib -L/home/simplicity/repos/cpp-darknet -L/usr/local/cuda/lib64 -o "testing"  ./src/testing.o ./src/yoloDarknet.o   -ldarknet-cpp-shared -lcudart -lcublas -lcurand -lopencv_imgcodecs -lopencv_imgproc -lopencv_core

**Comment by Qazi-Farjad** [comment]
g++ -L/usr/local/lib -L/home/simplicity/repos/cpp-darknet -L/usr/local/cuda/lib64 -o testing ./src/testing.o ./src/yoloDarknet.o -ldarknet-cpp-shared -lcudart -lcublas -lcurand -lopencv_imgcodecs -lopencv_imgproc -lopencv_core

**Comment by prabindh** [comment]
You are most likely having https://github.com/prabindh/darknet/issues/11

**Comment by Qazi-Farjad** [comment]
Ah, It works. Thank you.
